### DISCUSSION TITLE
Add custom app

### DISCUSSION THREAD

**Discussion by CorruptionVR** [DustinBrett/daedalOS#219] [Feature Requests]
Is it possible to add our own web apps to the OS? like, adding a website to open in a window using a shortcut.

**Comment by CorruptionVR** [comment]
nevermind, I found out that you can create a .URL file with the template 

[InternetShortcut]
BaseURL=
IconFile=
URL=

**Comment by CorruptionVR** [comment]
wait, how do I remove the browser UI?

**Comment by DustinBrett** [comment]
The shortcuts are for launching existing apps and passing them a "url" path within daedalOS. What you are asking for sounds more like 3rd party app support / iframe apps which is indeed something I have planned. But currently you'd need to make a custom app to do what you want most likely.

**Reply by CorruptionVR** [reply]
so, I just need to make a .url file?

**Reply by DustinBrett** [reply]
Well it depends what you want. If you want to launch an existing app then yes. But if you want to launch an app that is custom in some way, that doesn't exist yet. To do that you'd need to change the code. I plan to make an app that is generic in this way to allow external or 3rd party scripts. But it doesn't yet exist.

**Reply by CorruptionVR** [reply]
what I want is to add a "browser app", which yes, means that I want to make a window with an iframe set to a certain URL without the browser UI shown.

**Reply by DustinBrett** [reply]
I see. Yes currently that is not possible. But an iframe type of generic app that isn't Browser focused is planned for the future.

**Reply by Yakito** [reply]
Hi! the work here is amazing! congrats. Was this "generic" app implemented already? Thanks again!

**Reply by DustinBrett** [reply]
> Hi! the work here is amazing! congrats. Was this "generic" app implemented already? Thanks again!

Thanks very much! Sadly I have not yet got to the 3rd party app support. It's still possible to make apps via the source code, but packages apps that others make I have not yet created a good way to do this. I think about it often and 2024 it will for sure happen.
